File: sdkwork-backend-sdk-swift/Sources/API/ChannelResourceApi.swift
import Foundation

public class ChannelResourceApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing channel resource
    public func update(body: PlusChannelResourceForm) async throws -> PlusApiResultPlusChannelResourceVO? {
        let response = try await client.put(ApiPaths.backendPath("/channel/resource"), body: body)
        return response as? PlusApiResultPlusChannelResourceVO
    }

    /// Create a new channel resource
    public func create(body: PlusChannelResourceForm) async throws -> PlusApiResultPlusChannelResourceVO? {
        let response = try await client.post(ApiPaths.backendPath("/channel/resource"), body: body)
        return response as? PlusApiResultPlusChannelResourceVO
    }

    /// Get channel resources by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusChannelResourceVO? {
        let response = try await client.post(ApiPaths.backendPath("/channel/resource/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusChannelResourceVO
    }

    /// Get all channel resources
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusChannelResourceVO? {
        let response = try await client.post(ApiPaths.backendPath("/channel/resource/list/all"), body: body)
        return response as? PlusApiResultListPlusChannelResourceVO
    }

    /// Get a channel resource by ID
    public func getById(id: String) async throws -> PlusApiResultPlusChannelResourceVO? {
        let response = try await client.get(ApiPaths.backendPath("/channel/resource/\(id)"))
        return response as? PlusApiResultPlusChannelResourceVO
    }

    /// Delete a channel resource
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/channel/resource/\(id)"))
        return response as? PlusApiResultBoolean
    }
}

File: sdkwork-backend-sdk-csharp/Api/ChannelResourceApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class ChannelResourceApi
    {
        private readonly HttpClient _client;

        public ChannelResourceApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing channel resource
        /// </summary>
        public async Task<PlusApiResultPlusChannelResourceVO?> UpdateAsync(PlusChannelResourceForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusChannelResourceVO>(ApiPaths.BackendPath("/channel/resource"), body);
        }

        /// <summary>
        /// Create a new channel resource
        /// </summary>
        public async Task<PlusApiResultPlusChannelResourceVO?> CreateAsync(PlusChannelResourceForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusChannelResourceVO>(ApiPaths.BackendPath("/channel/resource"), body);
        }

        /// <summary>
        /// Get channel resources by page
        /// </summary>
        public async Task<PlusApiResultPagePlusChannelResourceVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusChannelResourceVO>(ApiPaths.BackendPath("/channel/resource/list"), body, query);
        }

        /// <summary>
        /// Get all channel resources
        /// </summary>
        public async Task<PlusApiResultListPlusChannelResourceVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusChannelResourceVO>(ApiPaths.BackendPath("/channel/resource/list/all"), body);
        }

        /// <summary>
        /// Get a channel resource by ID
        /// </summary>
        public async Task<PlusApiResultPlusChannelResourceVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusChannelResourceVO>(ApiPaths.BackendPath($"/channel/resource/{id}"));
        }

        /// <summary>
        /// Delete a channel resource
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/channel/resource/{id}"));
        }
    }
}
